### ISSUE TITLE
[Feature Request] Accounts - account groups grouping & API endpoint

### ISSUE THREAD

**Issue by lenisko** [RealDeviceMap/RealDeviceMap#431]
**Is your feature request related to a problem? Please describe.**
Most of the data is already shown on `/dashboard/accounts` page, but it lacks grouping by account groups. API could be useful for metrics and other stuff (like alerts on low accounts).

**Describe the solution you'd like**
- Add `groups` to `/dashboard/accounts` page
- Add new API endpoint for accounts, example schema:
```json
{
  "data": {
    "accounts": {
      "default": {
        "total": 300,
        "good": 100,
        "banned": 2,
        "warning": 1,
        "invalid": 0,
        "other": 0,
        "cooldown": 190,
        "spin_limit": 0
      },
      "iv_group": {
        "total": 300,
        "good": 100,
        "banned": 2,
        "warning": 1,
        "invalid": 0,
        "other": 0,
        "cooldown": 190,
        "spin_limit": 0
      }
    },
    "timestamp": 1666839630
  },
  "status": "ok"
}
```

**Comment by lenisko** [comment]
As for implementation

https://github.com/kbtbc/RealDeviceMap/blob/development/Sources/RealDeviceMapLib/Modell/Account.swift#L865-L893

All we need is to add `group` to select list and to `GROUP BY` statement. Like below:

```sql
SELECT
  `group`,
  level,
  COUNT(level) as total,
  SUM(
      (failed IS NULL AND first_warning_timestamp is NULL) OR
      (failed = 'GPR_RED_WARNING' AND warn_expire_timestamp IS NOT NULL AND
       warn_expire_timestamp != 0 AND warn_expire_timestamp <= UNIX_TIMESTAMP()) OR
      (failed = 'suspended' AND failed_timestamp <= UNIX_TIMESTAMP() - \(Account.suspendedPeriod))
  ) as good,
  SUM(failed IN('banned', 'GPR_BANNED')) as banned,
  SUM(first_warning_timestamp IS NOT NULL) as warning,
  SUM(
      (failed = 'GPR_RED_WARNING' AND warn_expire_timestamp IS NOT NULL AND
       warn_expire_timestamp != 0 AND warn_expire_timestamp > UNIX_TIMESTAMP())
  ) as inwarning,
  SUM(failed = 'invalid_credentials') as invalid_creds,
  SUM(failed = 'suspended') as suspended,
  SUM(failed = 'suspended'
      AND failed_timestamp > UNIX_TIMESTAMP() - \(Account.suspendedPeriod)) as insuspended,
  SUM(
    last_encounter_time IS NOT NULL AND UNIX_TIMESTAMP() -
    CAST(last_encounter_time AS SIGNED INTEGER) < 7200
  ) as cooldown,
  SUM(spins >= 1000) as spin_limit
FROM account
GROUP BY `group`, level
ORDER BY level DESC
```

And maybe add a select to form with `All, group1, group2, ...`.
Where `All` would sum all groups (to display exactly the same table as we have now) and all others would display levels only for selected group.

As for API I guess we could follow the pattern and return object of `group.level.type=counter` same way we have in table.

**Comment by versx** [comment]
> As for implementation
> 
> https://github.com/kbtbc/RealDeviceMap/blob/development/Sources/RealDeviceMapLib/Modell/Account.swift#L865-L893
> 
> All we need is to add `group` to select list and to `GROUP BY` statement. Like below:
> 
> ```sql
> SELECT
>   `group`,
>   level,
>   COUNT(level) as total,
>   SUM(
>       (failed IS NULL AND first_warning_timestamp is NULL) OR
>       (failed = 'GPR_RED_WARNING' AND warn_expire_timestamp IS NOT NULL AND
>        warn_expire_timestamp != 0 AND warn_expire_timestamp <= UNIX_TIMESTAMP()) OR
>       (failed = 'suspended' AND failed_timestamp <= UNIX_TIMESTAMP() - \(Account.suspendedPeriod))
>   ) as good,
>   SUM(failed IN('banned', 'GPR_BANNED')) as banned,
>   SUM(first_warning_timestamp IS NOT NULL) as warning,
>   SUM(
>       (failed = 'GPR_RED_WARNING' AND warn_expire_timestamp IS NOT NULL AND
>        warn_expire_timestamp != 0 AND warn_expire_timestamp > UNIX_TIMESTAMP())
>   ) as inwarning,
>   SUM(failed = 'invalid_credentials') as invalid_creds,
>   SUM(failed = 'suspended') as suspended,
>   SUM(failed = 'suspended'
>       AND failed_timestamp > UNIX_TIMESTAMP() - \(Account.suspendedPeriod)) as insuspended,
>   SUM(
>     last_encounter_time IS NOT NULL AND UNIX_TIMESTAMP() -
>     CAST(last_encounter_time AS SIGNED INTEGER) < 7200
>   ) as cooldown,
>   SUM(spins >= 1000) as spin_limit
> FROM account
> GROUP BY `group`, level
> ORDER BY level DESC
> ```
> 
> And maybe add a select to form with `All, group1, group2, ...`. Where `All` would sum all groups (to display exactly the same table as we have now) and all others would display levels only for selected group.
> 
> As for API I guess we could follow the pattern and return object of `group.level.type=counter` same way we have in table.

I agree with adding the select element to choose between specific groups or all. Just brainstorming but perhaps:
- `All` (everything regardless of group like you mentioned)
- `All Groups` / `Grouped` (include accounts with a group specified. DataTable has support for grouped table rows)
- `No Groups` / `None` (include accounts with no group specified)
- `Group1`
etc..

Bonus points for adding multiple selection support for the select element.
